scope_registry_test.go

```diff
@@ -143,10 +143,16 @@ func TestNewTestStatsReporterManyScopes(t *testing.T) {
 
 func TestForEachScopeConcurrent(t *testing.T) {
 	root := newRootScope(ScopeOptions{Prefix: "", Tags: nil}, 0)
+	quit := make(chan bool)
 	go func() {
 		for {
-			hello := root.Tagged(map[string]string{"a": "b"}).Counter("hello")
-			hello.Inc(1)
+			select {
+			case <-quit:
+				return
+			default:
+				hello := root.Tagged(map[string]string{"a": "b"}).Counter("hello")
+				hello.Inc(1)
+			}
 		}
 	}()
```

Comment:
Even though we have partial synchronization here (send/recv across an unbuffered channel), it only covers the recv itself and the function could be preempted before exiting (particularly if more logic is added in the future), so we should still ensure that it exits before the test completes. Even just something like:

```go
var (
  root = newRootScope(/* ... */)
  quit = make(chan bool)
  done = make(chan struct{})
)

go func() {
  defer close(done)

  for {
    select {
    case <-quit:
      return
    default:
      // ...
    }
  }
}()

// ...

var c Counter
for {
  // ...
}

<-done
```

(Applies both here and the snapshot case below.)

Reply:
done